### PR TITLE
Fixes for CI wheels

### LINKED ISSUE #450: update+fix Cython / ci-buildwheels
Effectively, a placeholder for a bunch of tasks
```[tasklist]
### Tasks
- [ ] https://github.com/SciTools/cf-units/issues/389
- [ ] #432
- [ ] #438
- [ ] #413
- [ ] #443
```

**NOTE:** I've relegated all those individual tasks from "Backlog" to "No Status" on the board
This simplifies the board, and I think they are probably cross-coupled (fixing one may fix others)
**So..** I think we first fix the basic build + then see what else still needs doing

### PR DESCRIPTION
**Pull request by HGWright**

### CODE REVIEW

File: .github/workflows/ci-wheels.yml [hunk 1]
```diff
@@ -32,36 +32,42 @@ jobs:
...
     - name: "Building ${{ matrix.os }} (${{ matrix.arch }}) wheels"
-      uses: pypa/cibuildwheel@v2.16.5
+      uses: pypa/cibuildwheel@v2.20.0
       env:
-        CIBW_SKIP: "cp36-* cp37-* cp38-* cp312-* pp* *-musllinux*"
+        CIBW_SKIP: "cp39-* cp313-* pp* *-musllinux*"
         CIBW_ARCHS: ${{ matrix.arch }}
         CIBW_BUILD_FRONTEND: build
         CIBW_MANYLINUX_X86_64_IMAGE: manylinux2014
+        CIBW_ENVIRONMENT: MACOS_DEPLOYMENT_TARGET=14.0
```
Comment (ESadek-MO) [inline .github/workflows/ci-wheels.yml]: comment this for the time being?

File: requirements/cf-units.yml [hunk 2]
```diff
@@ -8,12 +8,12 @@ dependencies:
   - setuptools >=45
   - setuptools_scm >=7.0
   - cython
-  - numpy
+  - numpy==1.26.4
```
Comment (ESadek-MO) [inline requirements/cf-units.yml]: does this still need to be pinned?

File: .github/workflows/ci-wheels.yml [hunk 3]
```diff
@@ -32,36 +32,42 @@ jobs:
...
     - uses: actions/checkout@v4
       with:
         fetch-depth: 0
 
     - name: "Building ${{ matrix.os }} (${{ matrix.arch }}) wheels"
-      uses: pypa/cibuildwheel@v2.16.5
+      uses: pypa/cibuildwheel@v2.20.0
       env:
-        CIBW_SKIP: "cp36-* cp37-* cp38-* cp312-* pp* *-musllinux*"
+        CIBW_SKIP: "cp39-* cp313-* pp* *-musllinux*"
```
Comment (ESadek-MO) [inline .github/workflows/ci-wheels.yml:66]: do we need 3.12?

**Review by ESadek-MO — COMMENTED** [review]
Couple questions, but generally looks good

File: .github/workflows/ci-wheels.yml (hunk 3 above)
Comment (HGWright) [reply .github/workflows/ci-wheels.yml:66]: yea i like showing that we can build the wheels on 3.12 even if we are not testing it for now

Comment (ESadek-MO) [reply .github/workflows/ci-wheels.yml:66]: yeah cool no drama

**Review by ESadek-MO — APPROVED** [review]
LGTM!

File: tox.ini [hunk 4]
```diff
@@ -4,7 +4,7 @@ requires =
     tox-run-command
  
  
-[testenv:py{39,310,311}-lock]
+[testenv:py{310,311,312}-lock]
```
Comment (trexfeathers) [inline tox.ini:7]: There are some missing corresponding changes. This is creating some confusing conflicts with #423. I could raise a pull request to fix these if you're OK with that?

https://github.com/SciTools/cf-units/blob/8a82027f25c97cfe531ff68e1b0ec09b1c0b2fa0/setup.cfg#L8-L12

https://github.com/SciTools/cf-units/blob/8a82027f25c97cfe531ff68e1b0ec09b1c0b2fa0/setup.cfg#L45-L46

https://github.com/SciTools/cf-units/blob/8a82027f25c97cfe531ff68e1b0ec09b1c0b2fa0/.github/workflows/ci-locks.yml#L28-L30

https://github.com/SciTools/cf-units/blob/8a82027f25c97cfe531ff68e1b0ec09b1c0b2fa0/.github/workflows/ci-tests.yml#L33-L42

https://github.com/SciTools/cf-units/blob/8a82027f25c97cfe531ff68e1b0ec09b1c0b2fa0/pyproject.toml#L50

https://github.com/SciTools/cf-units/blob/8a82027f25c97cfe531ff68e1b0ec09b1c0b2fa0/tox.ini#L36-L45

**Pull request merged**